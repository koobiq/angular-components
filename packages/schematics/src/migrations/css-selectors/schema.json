{
    "$schema": "http://json-schema.org/schema",
    "$id": "koobiq-components-css-selectors",
    "title": "koobiq-components css-selectors schematic",
    "type": "object",
    "properties": {
        "project": {
            "type": "string",
            "description": "Name of the project",
            "$default": {
                "$source": "projectName"
            }
        },
        "fix": {
            "type": "boolean",
            "description": "Whether overwrite icons prefixes or simply inform in console",
            "default": false
        },
        "stylesExt": {
            "type": "string",
            "description": "Use this option to specify styles extension preferred in your project",
            "default": "scss"
        }
    }
}
