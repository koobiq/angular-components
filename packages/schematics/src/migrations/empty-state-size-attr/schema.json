{
    "$schema": "http://json-schema.org/schema",
    "$id": "koobiq-components-empty-state-size-attr",
    "title": "koobiq-components empty-state-size-attr schematic",
    "type": "object",
    "properties": {
        "project": {
            "type": "string",
            "description": "Name of the project",
            "$default": {
                "$source": "projectName"
            }
        }
    }
}
