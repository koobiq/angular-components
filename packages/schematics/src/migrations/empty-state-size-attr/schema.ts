export interface Schema {
    /**
     * Name of the project where Koobiq library should be installed
     */
    project?: string;
}
