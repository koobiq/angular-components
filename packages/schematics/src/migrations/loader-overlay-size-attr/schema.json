{
    "$schema": "http://json-schema.org/schema",
    "$id": "koobiq-components-loader-overlay-size-attr",
    "title": "koobiq-components loader-overlay-size-attr schematic",
    "type": "object",
    "properties": {
        "project": {
            "type": "string",
            "description": "Name of the project",
            "$default": {
                "$source": "projectName"
            }
        }
    }
}
