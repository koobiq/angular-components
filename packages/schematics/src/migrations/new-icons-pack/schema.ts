export interface Schema {
    /**
     * Whether overwrite icons prefixes or simply inform in console
     */
    fix: boolean;
    /**
     * Use this option to specify styles extension preferred in your project
     */
    stylesExt: string;
    /**
     * Name of the project where Koobiq library should be installed
     */
    project?: string;
}
