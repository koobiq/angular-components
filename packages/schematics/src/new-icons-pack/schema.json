{
    "$schema": "http://json-schema.org/schema",
    "$id": "koobiq-components-new-icons-pack",
    "title": "koobiq-components new-icons-pack schematic",
    "type": "object",
    "properties": {
        "project": {
            "type": "string",
            "description": "Name of the project",
            "$default": {
                "$source": "projectName"
            }
        },
        "fix": {
            "type": "boolean",
            "description": "Whether overwrite icons prefixes or simply inform in console",
            "default": false
        },
        "stylesExt": {
            "type": "string",
            "description": "Use this option to specify styles extension preferred in your project",
            "default": "scss"
        }
    }
}
