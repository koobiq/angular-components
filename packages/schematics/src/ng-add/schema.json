{
    "$schema": "http://json-schema.org/schema",
    "$id": "koobiq-components-ng-add",
    "title": "koobiq-components ng-add schematic",
    "type": "object",
    "properties": {
        "project": {
            "type": "string",
            "description": "Name of the project",
            "$default": {
                "$source": "projectName"
            }
        }
    }
}
