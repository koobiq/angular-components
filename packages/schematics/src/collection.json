{
    "$schema": "../../../node_modules/@angular-devkit/schematics/collection-schema.json",
    "schematics": {
        "ng-add": {
            "description": "Adds @koobiq/components library to the project and installs dependencies",
            "factory": "./ng-add/index",
            "schema": "./ng-add/schema.json",
            "aliases": ["install"]
        },
        "new-icons-pack": {
            "description": "Updates prefixes and names of changed/removed icons",
            "factory": "./migrations/new-icons-pack/index",
            "schema": "./migrations/new-icons-pack/schema.json"
        },
        "css-selectors": {
            "description": "Updates prefixes and names of changed/removed icons",
            "factory": "./migrations/css-selectors/index",
            "schema": "./migrations/css-selectors/schema.json"
        },
        "deprecated-icons": {
            "description": "Updates prefixes and names of changed/removed icons",
            "factory": "./migrations/deprecated-icons/index",
            "schema": "./migrations/deprecated-icons/schema.json"
        }
    }
}
