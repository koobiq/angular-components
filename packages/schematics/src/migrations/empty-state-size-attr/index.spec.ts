import { ProjectDefinitionCollection } from '@angular-devkit/core/src/workspace';
import { Tree } from '@angular-devkit/schematics';
import { SchematicTestRunner } from '@angular-devkit/schematics/testing';
import { ProjectDefinition, WorkspaceDefinition } from '@schematics/angular/utility';
import { getWorkspace } from '@schematics/angular/utility/workspace';
import * as path from 'path';
import { createTestApp } from '../../utils/testing';
import { Schema } from './schema';

const collectionPath = path.join(__dirname, '../../collection.json');
const SCHEMATIC_NAME = 'empty-state-size-attr';

const getProjectContentPaths = (project: ProjectDefinition) => {
    return {
        templatePath: `/${project.root}/src/app/app.component.html`,
        tsPath: `/${project.root}/src/app/app.component.ts`,
        stylesPath: `/${project.root}/src/styles.scss`
    };
};

describe(SCHEMATIC_NAME, () => {
    let runner: SchematicTestRunner;
    let appTree: Tree;
    let projects: ProjectDefinitionCollection;
    let workspace: WorkspaceDefinition;

    beforeEach(async () => {
        runner = new SchematicTestRunner('schematics', collectionPath);
        appTree = await createTestApp(runner, { style: 'scss' });
        workspace = await getWorkspace(appTree);
        projects = workspace.projects as unknown as ProjectDefinitionCollection;
    });

    it('should run migration for specified project', async () => {
        const [firstProjectKey] = projects.keys();

        await runner.runSchematic(SCHEMATIC_NAME, { project: firstProjectKey } satisfies Schema, appTree);
    });

    it('should run migration for external html', async () => {
        const [firstProjectKey] = projects.keys();
        const { templatePath } = getProjectContentPaths(projects.get(firstProjectKey)!);

        const template =
            '<div>' +
            '<kbq-empty-state [big]="false"></kbq-empty-state>' +
            '<kbq-empty-state [big]="true"></kbq-empty-state>' +
            '</div>';

        appTree.overwrite(templatePath, template);

        const updatedTree = await runner.runSchematic(
            SCHEMATIC_NAME,
            { project: firstProjectKey } satisfies Schema,
            appTree
        );

        expect(updatedTree.read(templatePath)?.toString()).toMatchSnapshot(`project ${firstProjectKey}: after changes`);
    });

    it('should throw message if replaced attr value is not static', async () => {
        const warnSpy = jest.spyOn(console, 'warn').mockImplementation();
        const [firstProjectKey] = projects.keys();
        const { templatePath } = getProjectContentPaths(projects.get(firstProjectKey)!);

        // Set up the template with a non-static attribute value
        const nonStaticTemplate = '<div><kbq-empty-state [big]="VARIABLE"></kbq-empty-state></div>';

        appTree.overwrite(templatePath, nonStaticTemplate);
        const templateBeforeUpdate = appTree.read(templatePath)?.toString();

        const updatedTree = await runner.runSchematic(
            SCHEMATIC_NAME,
            { project: firstProjectKey } satisfies Schema,
            appTree
        );

        expect(templateBeforeUpdate).toBe(updatedTree.read(templatePath)?.toString());
        expect(warnSpy.mock.calls.some(([msg]) => msg.includes(templatePath))).toBe(true);
    });
});
