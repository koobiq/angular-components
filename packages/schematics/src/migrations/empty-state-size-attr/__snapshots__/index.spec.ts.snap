// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`empty-state-size-attr should run migration for external html: project app: after changes 1`] = `"<div><kbq-empty-state size="compact"></kbq-empty-state><kbq-empty-state size="big"></kbq-empty-state></div>"`;
